src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/SSHUserPrivateKeySnapshotTaker.java

```diff
@@ -0,0 +1,33 @@
+package com.cloudbees.jenkins.plugins.sshcredentials.impl;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHUserPrivateKey;
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey.DirectEntryPrivateKeySource;
+import com.cloudbees.plugins.credentials.CredentialsSnapshotTaker;
+
+import hudson.Extension;
+import hudson.util.Secret;
+
+@Extension
+public class SSHUserPrivateKeySnapshotTaker extends CredentialsSnapshotTaker<SSHUserPrivateKey> {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Class<SSHUserPrivateKey> type() {
+        return SSHUserPrivateKey.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SSHUserPrivateKey snapshot(SSHUserPrivateKey credentials) {
+        if (credentials instanceof BasicSSHUserPrivateKey) {
```

Comment:
Could be more defensive
```suggestion
        if (credentials instanceof BasicSSHUserPrivateKey && ((BasicSSHUserPrivateKey) credentials).getPrivateKeySource() instanceof DirectEntryPrivateKeySource) {
```
just in case some other plugin (unwisely) implements a new `PrivateKeySource`.